__tests__/commands/add.js

```diff
@@ -609,6 +609,27 @@ test.concurrent('upgrade scenario 2 (with sub dependencies)', (): Promise<void>
   });
 });
 
+test.concurrent('install another fork of an existing package', (): Promise<void> => {
+  // When installing a package with the same name as an existing one but from a different repo,
+  // the old one should be replaced with the new one in the lock file.
+  const firstSource = 'https://github.com/davidreis97/example-yarn-package.git';
+  const secondSource = 'https://github.com/yarnpkg/example-yarn-package.git';
```

Comment:
To avoid making requests in the ci, can you add the tarball cache to `__tests__/fixtures/request-cache/GET/github.com`